lib/rex/proto/http/client.rb

```diff
@@ -271,26 +271,30 @@ def send_auth(res, opts, t, persist)
 
     return res if opts['username'].nil? or opts['username'] == ''
     supported_auths = res.headers['WWW-Authenticate']
-    if supported_auths.include? 'Basic'
+
+    # if several providers are available, the client may want one in particular
+    preferred_auth = opts['preferred_auth']
+
+    if supported_auths.include?('Basic') && (preferred_auth.nil? || preferred_auth == 'Basic')
```

Comment:
I don't see how - there are multiple conditions in the `if`s... doesn't really seem like an improvement to force it into a `case` block.